// MIT License
//
// Copyright (c) 2021 EASE lab
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in all
// copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.vhive.stacking";
option java_outer_classname = "stacking";
option go_package = "tests/stacking-training/proto";

package stacking;

service Trainer {
    rpc Train(TrainRequest) returns (TrainReply) {}
}

message TrainRequest {
    // s3, xdt
    bytes dataset = 1;
    string dataset_key = 2;
    bytes model_config = 3;
    int64 count = 4;
    float sample_rate = 5;
}

message TrainReply {
    // s3, xdt
    bytes model = 1;
    string model_key = 2;
    string pred_key = 3;
    float score = 4;
    bytes params = 5;
}

